pkg/controllers/restore/controller.go

```diff
@@ -370,6 +370,12 @@ func (h *handler) generateDependencyGraph(ownerToDependentsList map[string][]res
 		}
 
 		customize(&resourceData)
+		if resourceData.GetKind() == "Deployment" && namespace == "cattle-system" {
```

Comment:
This check can be moved all the way as the first thing we do in this loop.

Reply:
moved.